test/functional/p2p_instantsend.py

```diff
@@ -54,11 +56,11 @@ def test_block_doublespend(self):
         for node in connected_nodes:
             self.wait_for_instantlock(is_id, node)
         # send doublespend transaction to isolated node
-        isolated.sendrawtransaction(dblspnd_tx['hex'])
+        dblspnd_txid = isolated.sendrawtransaction(dblspnd_tx['hex'])
         # generate block on isolated node with doublespend transaction
-        self.bump_mocktime(1)
-        isolated.generate(1)
-        wrong_block = isolated.getbestblockhash()
+        self.bump_mocktime(600)
```

Comment:
Maybe test also that it is **exactly** 600 seconds?
```
diff --git a/test/functional/p2p_instantsend.py b/test/functional/p2p_instantsend.py
index 55b43160c8..8e6c529141 100755
--- a/test/functional/p2p_instantsend.py
+++ b/test/functional/p2p_instantsend.py
@@ -58,7 +58,10 @@ class InstantSendTest(DashTestFramework):
         # send doublespend transaction to isolated node
         dblspnd_txid = isolated.sendrawtransaction(dblspnd_tx['hex'])
         # generate block on isolated node with doublespend transaction
-        self.bump_mocktime(600)
+        self.bump_mocktime(599)
+        wrong_early_block = isolated.generate(1)[0]
+        assert not "confirmation" in isolated.getrawtransaction(dblspnd_txid, 1)
+        self.bump_mocktime(1)
         wrong_block = isolated.generate(1)[0]
         assert_equal(isolated.getrawtransaction(dblspnd_txid, 1)["confirmations"], 1)
         # connect isolated block to network
```

Reply:
Good idea! Added in [d92085e](https://github.com/dashpay/dash/pull/5444/commits/d92085ee97689d9ca41514e7f3546581ba61d76d)